python/telemetry/snet.py

```diff
@@ -168,10 +167,57 @@ def _decode(self, obj, context, path=None):
     'TargetData_ControlIsActive' / Int8ul  # flag, really
     )
 
+# Mailbox in SNET satellites
+# There are three main fields defining the Mailbox inside SNET satellites:
+#
+# Name        ByteLength  ParamType   TargetType    Interpretation
+# SysTime     4           uint32_t    float         scale: 0,001 unit: s
+# MemDataType 1           uint8_t     uint8_t       values: 0, 1, 2, 3, 4, 5
+# MemData32   32          uint8_t     Array         format: Array
+#
+# Notes:
+# * `MemDataType` refers to the slot of memory in the satellite where
+# the information is stored, which slot of memory.
+# * `MemData32` is an Array of bytes, where each byte corresponds to an ascii
+# character code. The transmission of words is done by Big-Endian order.
 
-snet = Struct(
-    'header' / SNETFrameHeader,
-    'telemetry' / Switch(lambda c: (c.header.FCIDMajor, c.header.FCIDSub), {
+
+class MemData32Adapter(Adapter):
+    def _encode(self, obj, context, path=None):
+        return bytes([x << 1 for x in bytes(
+            (obj.upper() + ' '*8)[:8], encoding='ascii')])
+
+    def _decode(self, obj, context, path=None):
+        # do we need to do right shift here?
+        return str(bytes([x for x in bytes(
+            (obj.upper() + ' '*8)[:8], encoding='ascii')]),
+                   encoding='ascii').strip()
+
+
+MemData32 = MemData32Adapter(BitsInteger(256))
+
+MailboxReceiveTelemetry = Struct(
+    'SysTime' / BitsInteger(32),
```

Comment:
Why `BitsInteger(32)` instead of something like `Int32sb`?

Reply:
no strong reason why, I have being using `Bits*` mostly, but `Int32sb` works too